am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/defaultrolesetup/AddResourceDefinitionIntegrationTest.java

```diff
@@ -0,0 +1,40 @@
+package integration.uk.gov.hmcts.reform.amlib.defaultrolesetup;
+
+import integration.uk.gov.hmcts.reform.amlib.base.IntegrationBaseTest;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
+class AddResourceDefinitionIntegrationTest extends IntegrationBaseTest {
+
+    @Test
+    void whenAddResourceDefinitionIsCalledAddNewEntryIntoDatabase() {
+        defaultRoleService.addService(SERVICE_NAME, "");
+
+        defaultRoleService.addResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME);
+        assertThat(countResources(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME)).isEqualTo(1);
+    }
+
+    @Test
+    void whenCallIsMadeDuplicatingExistingResourceUpdatesExistingEntry() {
+        defaultRoleService.addService(SERVICE_NAME, "");
+
+        defaultRoleService.addResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME);
+        defaultRoleService.addResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME);
+
+        assertThat(countResources(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME)).isEqualTo(1);
+    }
+
+    @Test
+    void canDeleteResourceDefinitionFromTable() {
```

Comment:
I personally don't mind. The class name could be updated to relfect that all tests to do with resources are in one class? Or it can be all separate. What are your thoughts?